scipy/signal/signaltools.py

```diff
@@ -892,41 +892,84 @@ def _prod(iterable):
     return product
 
 
-def _fftconv_faster(x, h, mode):
+def _conv_ops(x_shape, h_shape, mode):
+    x_size, h_size = _prod(x_shape), _prod(h_shape)
+    if mode == "full":
+        out_shape = [n + k - 1 for n, k in zip(x_shape, h_shape)]
+    elif mode == "valid":
+        out_shape = [abs(n - k) + 1 for n, k in zip(x_shape, h_shape)]
+    elif mode == "same":
+        out_shape = x_shape
+    else:
+        raise ValueError("Acceptable mode flags are 'valid',"
+                         " 'same', or 'full', not mode={}".format(mode))
+
+    s1, s2 = x_shape, h_shape
+    if len(x_shape) == 1:
+        s1, s2 = s1[0], s2[0]
+        if mode == "full":
+            direct_ops = s1 * s2
+        elif mode == "valid":
+            direct_ops = (s2 - s1 + 1) * s1 if s2 >= s1 else (s1 - s2 + 1) * s2
+        elif mode == "same":
+            direct_ops = s1 * s2 if s1 <= s2 else s1 * s2 - (s2 // 2) * ((s2 + 1) // 2)
```

Comment:
Small bug: it should be `s1 < s2`.

Reply:
You're right. I've fixed this. I misread https://github.com/scipy/scipy/blob/3af8448114dd4a2755d513edb5a2272a338dc483/scipy/signal/signaltools.py#L952

(here, `(s1, s2) == (volume.size, kernel.size)`, so this condition can be rewritten as `s2 <= s1`, and the converse is `s1 < s2`).